nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -271,18 +271,29 @@ def _cleanup_workspace(self):
         with tempfile.TemporaryDirectory() as temp_dir:
             startup_dir = os.path.join(self.args.workspace, "startup")
             temp_start_up = os.path.join(temp_dir, "startup")
+            local_dir = os.path.join(self.args.workspace, "local")
+            temp_local_dir = os.path.join(temp_dir, "local")
             if os.path.exists(startup_dir):
                 shutil.move(startup_dir, temp_start_up)
+            if os.path.exists(local_dir):
+                shutil.move(local_dir, temp_local_dir)
+
             if os.path.exists(self.simulator_root):
                 shutil.rmtree(self.simulator_root)
+
             if os.path.exists(temp_start_up):
                 shutil.move(temp_start_up, startup_dir)
+            if os.path.exists(temp_local_dir):
+                shutil.move(temp_local_dir, local_dir)
 
     def _setup_local_startup(self, log_config_file_path, workspace):
         local_dir = os.path.join(workspace, "local")
         startup = os.path.join(workspace, "startup")
         os.makedirs(local_dir, exist_ok=True)
         shutil.copyfile(log_config_file_path, os.path.join(local_dir, WorkspaceConstants.LOGGING_CONFIG))
+        workspace_local = os.path.join(self.simulator_root, "local")
+        if os.path.exists(workspace_local):
+            shutil.copytree(workspace_local, local_dir, dirs_exist_ok=True)
```

Comment:
So we want the content from "simulator_root"/"local" to overwrite "simulator_root" / p / "local"?
For simulator, are all sites share the same "simulator_root"/"local" folder hence same config in "simulator_root"/"local"?

Reply:
Yes. For simulator there's no fix number, or fix name of the clients. Each run will have its own number of clients.